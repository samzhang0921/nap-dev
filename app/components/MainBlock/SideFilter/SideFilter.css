.sideFilter {
max-width: 183px;
padding: 13px 0 0;
position: relative;
}
@media (max-width: 949px) {
.sideFilter {
    width: 100%;
    display: none;
  }
}
