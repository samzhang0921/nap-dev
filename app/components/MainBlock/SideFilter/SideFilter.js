import React from 'react';
import DesignerFilter from './DesignerFilter/DesignerFilter';
import CategoryFilter from './CategoryFilter/CategoryFilter';
import ColorFilter from './ColorFilter/ColorFilter';
import style from './SideFilter.css';
import PropTypes from 'prop-types';

const SideFilter = props => {
  return (
    <div className={style.sideFilter}>

      <CategoryFilter updateCategory = {props.updateCategory}/>

      <DesignerFilter designerList={props.designerList} updateDesignerList={props.updateDesignerList} categoryRefresh={props.categoryRefresh}/>

      <ColorFilter updateColorList = {props.updateColorList} colorList ={props.colorList} categoryRefresh={props.categoryRefresh}/>

    </div>
  )
};
SideFilter.propTypes = {
  designerList:PropTypes.array,
  colorList:PropTypes.array,
  productViewChecked :PropTypes.bool,
  updateCategory : PropTypes.func,
  updateDesignerList : PropTypes.func,
  updateColorList : PropTypes.func,
  categoryRefresh : PropTypes.bool
};

export default SideFilter;
