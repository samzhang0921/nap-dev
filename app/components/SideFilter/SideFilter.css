.sideFilter {
  float: left;
width: 183px;
padding: 13px 0 0;
position: relative;
background-color: red;
}
