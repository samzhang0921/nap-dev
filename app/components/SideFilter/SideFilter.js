import React from 'react';
import DesignerFilter from '../DesignerFilter/DesignerFilter';
import CategoryFilter from '../CategoryFilter/CategoryFilter';
import ColorFilter from '../ColorFilter/ColorFilter';
import style from './SideFilter.css';

const SideFilter = props => {
  return (

    <div className={style.sideFilter}>

      <CategoryFilter updateCategory = {props.updateCategory}/>

      <DesignerFilter designerList={props.designerList}/>

      <ColorFilter/>

    </div>
  )
}

export default SideFilter;
