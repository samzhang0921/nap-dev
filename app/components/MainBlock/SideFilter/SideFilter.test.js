import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import SideFilter from './SideFilter';

it('should render a todo item ', () => {
  const tree = toJson(shallow(<SideFilter data={[]} />));
  expect(tree).toMatchSnapshot();
});

it('should match its snapshot with items', () => {
  const data = ['a', 'b', 'c'];
  const tree = renderer.create(
    <SideFilter data={data} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
