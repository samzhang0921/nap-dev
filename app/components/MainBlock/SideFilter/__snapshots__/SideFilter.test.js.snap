// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match its snapshot with items 1`] = `
<div
  className="sideFilter"
>
  <div
    className="CatFilterList"
  >
    <div
      className="filterItem"
      data-category-id={0}
      onClick={[Function]}
    >
      Clothing
    </div>
    <div
      className="filterItem"
      data-category-id={40}
      onClick={[Function]}
    >
      Beachwear
    </div>
    <div
      className="filterItem"
      data-category-id={48}
      onClick={[Function]}
    >
      Coat
    </div>
    <div
      className="filterItem"
      data-category-id={46}
      onClick={[Function]}
    >
      Dress
    </div>
    <div
      className="filterItem"
      data-category-id={44}
      onClick={[Function]}
    >
      Jackets
    </div>
    <div
      className="filterItem"
      data-category-id={55}
      onClick={[Function]}
    >
      Jeans
    </div>
    <div
      className="filterItem"
      data-category-id={42}
      onClick={[Function]}
    >
      Pants
    </div>
    <div
      className="filterItem"
      data-category-id={39}
      onClick={[Function]}
    >
      Top
    </div>
    <div
      className="filterItem"
      data-category-id={49}
      onClick={[Function]}
    >
      Wrap
    </div>
  </div>
  <div
    className="DesignerFilter"
  >
    Loading
  </div>
  <div
    className="ColorFilter"
  >
    Loading
  </div>
</div>
`;

exports[`should render a todo item  1`] = `
<div
  className="sideFilter"
>
  <CategoryFilter />
  <DesignerFilter />
  <ColorFilter />
</div>
`;
